### ISSUE TITLE
Add Modern Cloze Overlapper to ANKING note type

### ISSUE THREAD

**Issue by amatsune** [AnKing-VIP/AnKing-Note-Types#188]
#### Which Note Type (and version) is this about?

All notes that use Cloze deletions

#### Checklist

 - [x] I've verified that I use the latest version of the note type by redownloading it from AnkiWeb
 - [X] I've checked if anyone else suggested this feature before by looking through the issue reports.

#### What Anki version are you using?

```
24.04.1 Qt6
```

#### What Operating system are you using? (i.e. MacOS, Windows, Linux)

Windows 11

#### Problem case

Currently it's not possible to create real "overlapping clozes" in the Anking card format. I'm aware of the One-by-One cloze feature, but it lacks the option to hide the remaining cloze field (similar to the Image Occlusion hide one-guess one/hide all-guess one.

I often have a need for cards that share overlapping cloze contents but that only reveal one field in the front like this:

![image](https://github.com/AnKing-VIP/AnKing-Note-Types/assets/48121978/d9366342-0616-4b0f-8326-c041915303f6)

I achieved this result using the [Modern Cloze Overlapper (MCO)](https://github.com/nihil-admirari/modern-cloze-overlapper).

I tried to make a copy of the Anking card type and edit the script into the card format, but didn't manage to get it to work. I had a card with duplicated text:

Front:
![image](https://github.com/AnKing-VIP/AnKing-Note-Types/assets/48121978/3844a377-e7b5-4329-9628-6ca0aeb7198e)

Back:
![image](https://github.com/AnKing-VIP/AnKing-Note-Types/assets/48121978/c7966461-a687-4fe7-bccf-51a934659eb2)

I tried removing the part of the code that output the front/back text, and managed to remove the duplication issue, but that gave me another set of problems: I either got rid of the anking one-by-one function and formatting/styling (notice the difference in the shade of pink of the cloze field) or the MCO didn't do anything.

**Front:**

![image](https://github.com/AnKing-VIP/AnKing-Note-Types/assets/48121978/59399bc0-531a-402a-a7bf-07a496e54afc)
MCO single cloze (Card 6) - lost Anking formatting

![image](https://github.com/AnKing-VIP/AnKing-Note-Types/assets/48121978/590503e2-1ca3-4ee2-a580-73ec6241f9e5)
Anking O-b-O (card 6), all other clozes revealed

![image](https://github.com/AnKing-VIP/AnKing-Note-Types/assets/48121978/b13f1099-7cde-4515-9dde-a026d8ce6c3d)
MCO (Card 7) - lost Anking formatting, otherwise working as expected

![image](https://github.com/AnKing-VIP/AnKing-Note-Types/assets/48121978/9c9813cb-6edc-4935-95bd-f09bcdeda2ec)
Anking O-b-O (Card 7) working as expected

**Back:**

![image](https://github.com/AnKing-VIP/AnKing-Note-Types/assets/48121978/959ceb55-de41-477d-96fe-c5bca8a98d09)
MCO (card 6) lost Anking formatting

![image](https://github.com/AnKing-VIP/AnKing-Note-Types/assets/48121978/0fe9af51-0394-4963-bc5c-d440ace9bec4)
Anking (Card 6) all revealed by default

![image](https://github.com/AnKing-VIP/AnKing-Note-Types/assets/48121978/63a5efa8-a1f4-4bed-a524-e4a2802e8e23)
MCO (Card 7) no O-b-O - lost Anking formatting

![image](https://github.com/AnKing-VIP/AnKing-Note-Types/assets/48121978/8d111c47-bdcf-4567-9dfb-4a6c9455a964)
If I do click the toggle button, however, it re-establishes the one-by-one feature, though it defeats the purpose if it's revealed before I try to answer the card. The click to reveal function doesn't work, only the reveal next button.

![image](https://github.com/AnKing-VIP/AnKing-Note-Types/assets/48121978/82615cae-2de5-40a1-a601-f8c8396da04a)
Anking O-b-O (Card 7), two answers clicked on to reveal out of order - works as intended

I can't seem to find a way to reconcile the two features and get them to work together.

#### Solution

Integrate the function of the MCO card to the Anking note types (including the Before|After|OnlyContext|RevealAll|InactiveHints customization field).

Else, make the Anking note type at least compatible, for those who decide to manually add the required card formatting script.

Else, if possible, I'd settle for some guidance on how to create a custom workaround for me and others who might have an interest.

#### More information

I know I'm overreaching at this point but... since this issue is regarding overlapping clozes, if there was a way to create clozes that only partially overlap such as:

{{c1::((c3::This [[c2::is)) a ((c3::cloze}} deletion))]]

**Card 1**
_**[…]** deletion_

**Card 2**
_This **[…]**_

**Card 3**
_**[…]** a **[…]**_

Not sure how the syntax for that would have to work, so I used multiple punctuation marks to make them clear, though something like this would likely be optimal:

{{c1::{{c3::This {{c2::is::c3}} a {{c3::cloze::c1::hint}} deletion::c3::multi word hint}}::c2}}

**Card 1**
_**[hint]** deletion_

**Card 2**
_This **[…]**_

**Card 3**
_**[…]** a **[multi word hint]**_

If keeping the MCO Before|After|OnlyContext|RevealAll|InactiveHints field, a possible result would be something like:

{{c4::This {{c1::is a much {{c2::longer::c1::hint 1}} cloze::c2::hint 2}}::c4::hint 4}} deletion for {{c3::exemple::c3:: hint 3}}

**Card 1**
_This **[hint 1]** cloze deletion for exemple_

**Card 2**
_This is a much **[hint 2]** deletion for exemple_

**Card 3**
_This is a much longer cloze deletion for **[hint 3]**_

**Card 4**
_**[hint 4]** deletion for exemple_

 
_Some different settings would result in:_

**Card 1** (1,1,false,true,false)
_This **[hint 1]** cloze deletion for exemple_

**Card 1** (0,2,false,true,true)
_This **[hint 1]** cloze deletion for exemple_

**Card 2** (1,0,true,true,false)
_This is a much **[hint 2]** deletion for_

**Card 3** (1,1,false,true,false)
_This **[...]** cloze deletion for **[hint 3]**_

**Card 4** (0,0,false,true,true)
_**[hint 4]** deletion for **[hint 3]**_

Card 4 (1,1,false,true,true)
_**[hint 4]** deletion for exemple_

EDIT: Finished the suggestion and some formatting.

**Comment by AnKingMed** [comment]
@abdnh or @RisingOrange I think I discussed this idea with you in the past to make it possible to do a cloze overlapper via the note styling without needing the addon. Might be worth looking into this here?

**Comment by abdnh** [comment]
@AnKingMed I think you mentioned implementing something like this https://github.com/michalrus/anki-simple-cloze-overlapper?

**Comment by AnKingMed** [comment]
Yeah whatever you think is simplest and best

**Comment by AnKingMed** [comment]
Could we use one by one field?

**Comment by abdnh** [comment]
We can but that's unintuitive, since it's not tied to the one by one feature.

**Comment by abdnh** [comment]
@amatsune Does this sample deck work as expected for you? (Try it in a new profile)
[cloze overlapper.zip](https://github.com/user-attachments/files/16086489/cloze.overlapper.zip)

It's a basic implementation of cloze overlapper that works with the one-by-one feature.

**Comment by AnKingMed** [comment]
works really well for me!

**Comment by AnKingMed** [comment]
Works well for me! I think it'd be worth putting a description in this field if possible? I just realized that we can do that! We could add that to the one by one fields too? 

![CleanShot 2024-07-17 at 11 32 03](https://github.com/user-attachments/assets/853d5a9a-f879-4aeb-bb01-8af55e0d6c27)

Here's the text I used: `(number of leading clozes - #), (number of following clozes - #), (show all clozes - true/false)`
